os/board/stm32l4r9ai-disco/src/stm32_boot.c

```diff
@@ -73,6 +73,9 @@ void board_initialize(void)
     board_led_initialize();
 #endif
 
+  /* Initialize for flash access */
+  stm32l4_flash_initialize();
```

Comment:
Mostly, flash should be initialized after SPI, because flash uses spi as a bus.
It already has stm32_exflash_initialize, please check it and use it if possible.

Reply:
Otherwise, up_fliashinitialize in stm32_flash.c

Reply:
A semaphore initialized by stm32l4_flash_initiailize is used in up_progmem functions.
But in mtd_initialize,
#ifdef CONFIG_MTD_PROGMEM
  progmem_initialize -> common code
#else
  up_flashinitialize() -> board_specific
We cannot put stm32l4_flash_initiailize() in up_flashinitialize so I called this function in board_initialize before calling mtd_partition.